prov/sockets/src/sock_epoll.c

```diff
@@ -61,10 +61,14 @@
 #ifdef HAVE_EPOLL
 int sock_epoll_create(struct sock_epoll_set *set, int size)
 {
+	int ret;
 	set->size = size;
 	set->used = 0;
 	set->events = calloc(size, sizeof(struct epoll_event));
-	return set->fd = epoll_create(size);
+	ret = set->fd = epoll_create(size);
+	if (ret < 0)
+		free(set->events);
```

Comment:
I just checked, we free events on <code>sock_epoll_close</code> which is called by <code>sock_conn_map_destroy</code>, <code>sock_pe_init</code> and <code>sock_pe_finalize</code>. The above free will be called only if <code>sock_epoll_create</code> fails inside <code>sock_pe_init</code>. In this error path, <code>sock_epoll_close</code> is not called. So I think we don't have double free.